### PR TITLE
Rediseconomy compatibility

### PR DESCRIPTION
**Pull request by Emibergo02**
same as [https://github.com/WiIIiam278/HuskHomes2/pull/259#issue-1440377840](https://github.com/WiIIiam278/HuskHomes2/pull/259#issue-1440377840)

### CODE REVIEW

File: bukkit/build.gradle [hunk 1]
```diff
@@ -8,6 +8,7 @@ dependencies {
     implementation ('com.zaxxer:HikariCP:5.0.1') {
         exclude module: 'slf4j-api'
     }
+    implementation 'com.github.Emibergo02:RedisEconomy:2.1-SNAPSHOT'
```
Comment (WiIIiam278) [inline bukkit/build.gradle:11]: Why are we implementing it? This requires the user to have the plugin installed, yeah? So we don't need to bundle it with this jar. Change to compileOnly and remove the relocate.

File: bukkit/src/main/java/net/william278/husktowns/hook/EconomyHook.java [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package net.william278.husktowns.hook;
 
+import dev.unnm3d.rediseconomy.api.RedisEconomyAPI;
```
Comment (WiIIiam278) [inline bukkit/src/main/java/net/william278/husktowns/hook/EconomyHook.java:3]: This breaks the Vault economy support. The plugin will crash with a ClassDefNotFoundException as you're trying to import a class that may not exist.

File: bukkit/src/main/resources/plugin.yml [hunk 3]
```diff
@@ -27,7 +28,7 @@ commands:
     description: Create, manage & view details about towns
     usage: /town [command]
     permission: husktowns.command.town
-    aliases: [ t ]
+    aliases: [ t, village ]
```
Comment (WiIIiam278) [inline bukkit/src/main/resources/plugin.yml:31]: remove this :-)

File: bukkit/src/main/java/net/william278/husktowns/hook/EconomyHook.java [hunk 4]
```diff
@@ -15,6 +16,15 @@ public static boolean initialize() {
         if (!HuskTowns.getSettings().doEconomy) {
             return false;
         }
+        if (Bukkit.getPluginManager().getPlugin("RedisEconomy") != null) {
```
Comment (WiIIiam278) [inline bukkit/src/main/java/net/william278/husktowns/hook/EconomyHook.java:19]: Yeah, should move this to its' own class entirely.

**Review by WiIIiam278 — CHANGES_REQUESTED** [review]
This implementation sadly breaks Vault support :c Need to move this to its' own class.

File: bukkit/src/main/resources/plugin.yml (hunk 3 above)
Comment (Emibergo02) [reply bukkit/src/main/resources/plugin.yml]: k

**Review by WiIIiam278 — APPROVED** [review]
Nice work, good stuff with the refactor.

**Pull request merged**